libraries/SITL/SIM_Aircraft.cpp

```diff
@@ -737,4 +736,27 @@ void Aircraft::extrapolate_sensors(float delta_time)
     velocity_air_bf = dcm.transposed() * velocity_air_ef;
 }
 
+void Aircraft::update_external_payload(const struct sitl_input &input)
+{
+    external_payload_mass = 0;
+
+    // update sprayer
+    if (sitl->sprayer_sim.sprayer_enable) {
```

Comment:
Why not just `sitl->sprayer_sim.update()` here, passing through the sitl_input still?